### PR TITLE
BTS-619: Remove pure python stuff

### PR DESCRIPTION
**Pull request by chpedersen**

### CODE REVIEW

File: sharedbuffers/mapped_struct.py [hunk 1]
```diff
@@ -2429,7 +2302,6 @@ def _frozenset_eq(self, x):
 
     def __eq__(self, seq):
         if isinstance(self, proxied_frozenset):
-            return self._frozenset_eq(seq)
             return cython.cast(proxied_frozenset, self)._frozenset_eq(seq)
         else:
```
Comment (chpedersen) [inline sharedbuffers/mapped_struct.py:2280]: Ok, I don't remember why I added this, but the tests pass :shrug:

File: sharedbuffers/mapped_struct.py [hunk 2]
```diff
@@ -2567,7 +2439,6 @@ def __lt__(self, seq):
 
     def __le__(self, seq):
         if isinstance(self, proxied_frozenset):
-            return self._subset(seq, False)
             return cython.cast(proxied_frozenset, self)._subset(seq, False)
```
Comment (chpedersen) [inline sharedbuffers/mapped_struct.py:2416]: same here

File: sharedbuffers/mapped_struct.py [hunk 3]
```diff
@@ -1934,46 +1842,35 @@ def __getitem__(self, index):
...
-            return proxied_list_cmp(self, other) < 0
-
-        def __gt__(self, other):
-            if not islist(other):
-                raise NotImplementedError
-            return proxied_list_cmp(self, other) > 0
+    def __cmp__(self, other):
+        if not islist(other):
+            raise NotImplementedError
+        return proxied_list_cmp(self, other)
```
Comment (klaussfreire) [inline sharedbuffers/mapped_struct.py]: This was "if not cython.compiled"... shouldn't this have been deleted?

I don't see cython code for the comparison. Which is weird.

I think proxied_list should have a richcmp method like proxied_tuple... not sure why it isn't there.

**Review by marianoSecure — APPROVED** [review]

File: sharedbuffers/mapped_struct.py [hunk 4]
```diff
@@ -2675,8 +2546,6 @@ def __hash__(self):
                     if x == -1:
                         x = -2
                     self._hash = x
-            else:
-                self._hash = hash(tuple(iter(self)))
```
Comment (klaussfreire) [inline sharedbuffers/mapped_struct.py]: I think the else path is still needed

**Review by klaussfreire — CHANGES_REQUESTED** [review]

File: sharedbuffers/mapped_struct.py (hunk 4 above)
Comment (marianoSecure) [reply sharedbuffers/mapped_struct.py]: I think its being hashed correctly in line 2525, since implementation is always going to be cpython

Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py]: pypy is not cpython

**Review by klaussfreire — APPROVED** [review]

File: sharedbuffers/mapped_struct.py (hunk 4 above)
Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py]: And cython != cpython (one very significant letter of difference)

**Pull request merged**